api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -718,3 +719,27 @@ func (c *imageBuilder) createKanikoJobSpec(
 	}
 	return job, nil
 }
+
+func (c *imageBuilder) GetImageBuildingJobStatus(ctx context.Context, project mlp.Project, model *models.Model, version *models.Version) (models.ImageBuildingJobStatus, error) {
+	kanikoJobName := c.nameGenerator.generateBuilderJobName(project, model, version)
+
+	jobClient := c.kubeClient.BatchV1().Jobs(c.config.BuildNamespace)
+	job, err := jobClient.Get(ctx, kanikoJobName, metav1.GetOptions{})
+	if err != nil {
+		return models.ImageBuildingJobStatusUnknown, err
```

Comment:
what if the job is in pending state? Not started and before active. Or that is covered under active?

Reply:
just bit odd too that when there is error its unknown status, when there is no error its also unknown status

Reply:
-> what if the job is in pending state? Not started and before active. Or that is covered under active?
It's covered under active. Active is the number of pending and running pods. Ref: https://pkg.go.dev/k8s.io/api@v0.29.2/batch/v1#JobStatus